docs/enhancements/replace_broken_clusters.md

```diff
@@ -0,0 +1,69 @@
+# Automatically detect and replace broken Clusters
+
+## Summary
+
+Hibernating clusters often fail to transition to running state, but are assigned to ClusterClaims today. We would want to 
+validate a cluster can be successfully resumed and is running as expected before assigning it to a claim.
+
+## Motivation
+
+As we move towards being more proactive with having clusters ready as [hot spares](https://github.com/openshift/hive/pull/1434),
+we could come across instances where stale hibernated clusters could fail to resume and still be assigned to ClusterClaims.
+There's a need to detect broken clusters and replace them automatically.
+
+### Goals
+
+Always strive to assign a running cluster to a claim
+
+### Non Goals
+
+Fixing clusters (or attached claims) once a running cluster has already been assigned.
+
```

Comment:
I was hoping to understand how this change will affect the users.
1. Today we assign a cluster to claim and then it changes power state. the user watch the cluster running condition to see when the cluster is ready. How will this change this behavior?
2. How do we provide information to the claim owner about the progress being made.

Reply:
Good points. My take:
1. The cluster is ready as soon as the claim is assigned. (That's arguably better UX anyway.)
2. IMO nothing changes here: the `ClusterClaimPendingCondition` remains `True` with reason `NoClusters` ("No clusters in pool are ready to be claimed"). It's just that we've now got an additional stage during which this is the case. (I don't think it's a good idea to try to add another Reason here, like `ResumingCluster`, because we haven't done (and shouldn't do) anything to associate the claim with the cluster we're resuming.)

Also note that once `runningCount` is a thing and properly tuned, most/many claims will be assigned immediately anyway.